drivers/bebop_bus/BebopBus.cpp

```diff
@@ -0,0 +1,318 @@
+ *    notice, this list of conditions and the following disclaimer in
+ *    the documentation and/or other materials provided with the
+ *    distribution.
+ * 3. Neither the name PX4 nor the names of its contributors may be
+ *    used to endorse or promote products derived from this software
+ *    without specific prior written permission.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+ * "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+ * LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS
+ * FOR A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE
+ * COPYRIGHT OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT,
+ * INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING,
+ * BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS
+ * OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED
+ * AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT
+ * LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN
+ * ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+ * POSSIBILITY OF SUCH DAMAGE.
+ *
+ ****************************************************************************/
+
+#include <string.h>
+
+#include "DriverFramework.hpp"
+#include "BebopBus.hpp"
+
+#define BEBOP_REG_SET_ESC_SPEED 0x02
+#define BEBOP_REG_GET_OBS 0x20
+#define BEBOP_REG_START_BLDC 0x40
+#define BEBOP_REG_STOP_BLDC 0x60
+#define BEBOP_REG_TOGGLE_GPIO 0x4D
+#define BEBOP_REG_CLEAR_ERROR 0x80
+#define BEBOP_REG_PLAY_SOUND 0x82
+#define BEBOP_REG_GET_INFO 0xA0
+
+#define BEBOP_BUS_ERROR_NO 0x00
+#define BEBOP_BUS_ERROR_EEPROM 0x01
+#define BEBOP_BUS_ERROR_STALLED 0x02
+#define BEBOP_BUS_ERROR_PROPELLER_SECURETY 0x03
+#define BEBOP_BUS_ERROR_COMMUNICATION_LOST 0x04
+#define BEBOP_BUS_ERROR_LOW_BATTERY 0x09
+#define BEBOP_BUS_ERROR_LIPO_CELLS 0x0A
+#define BEBOP_BUS_ERROR_BLDC 0x0B
+
+// Rotation direction
+// front left: clockwise
+// front right: counterclockwise
+// back right: clockwise
+// back left: counterclockwise
+#define BEBOP_BLDC_RLRL 0b00001010
+// Rotation direction
+// front left: counterclockwise
+// front right: clockwise
+// back right: counterclockwise
+// back left: clockwise
+#define BEBOP_BLDC_LRLR 0b00000101 // 
+
+#define BEBOP_BLDC_RPM_MIN 3000
+#define BEBOP_BLDC_RPM_MAX 12200
+
+using namespace DriverFramework;
+
+int BebopBus::start()
+{
+	int result = I2CDevObj::start();
+
+	if (result < 0) {
+		DF_LOG_ERR("DevObj start failed");
+		DF_LOG_ERR("Unable to open the device path: %s", m_dev_path);
+		return result;
+	}
+
+	result = _setSlaveConfig(BEBOP_BUS_SLAVE_ADDRESS,
+				 BEBOP_BUS_BUS_FREQUENCY_IN_KHZ,
+				 BEBOP_BUS_TRANSFER_TIMEOUT_IN_USECS);
+
+	if (result < 0) {
+		DF_LOG_ERR("Could not set slave config");
+	}
+
+	/* Try to talk to the sensor. */
+	struct bebop_bus_info info;
+	result = _get_info(&info);
+
+	// TODO make some sanity checks
```

Comment:
Yes, unfortunately. The idea is, to check the driver version on the Bebop hardware, if it is compatible with the BebopBus driver in DriverFramework. At the moment, I am not aware of any issues, but this needs some further checking.